moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -355,14 +361,23 @@ bool JogCalcs::cartesianJogCalcs(geometry_msgs::TwistStamped& cmd, trajectory_ms
     Eigen::Vector3d translation_vector(cmd.twist.linear.x, cmd.twist.linear.y, cmd.twist.linear.z);
     Eigen::Vector3d angular_vector(cmd.twist.angular.x, cmd.twist.angular.y, cmd.twist.angular.z);
 
-    // We solve (planning_frame -> base -> cmd.header.frame_id)
-    // by computing (base->planning_frame)^-1 * (base->cmd.header.frame_id)
-    const auto tf_planning_to_cmd_frame =
-        kinematic_state_->getGlobalLinkTransform(parameters_.planning_frame).inverse() *
-        kinematic_state_->getGlobalLinkTransform(cmd.header.frame_id);
-
-    translation_vector = tf_planning_to_cmd_frame.linear() * translation_vector;
-    angular_vector = tf_planning_to_cmd_frame.linear() * angular_vector;
+    // If the incoming frame is empty or is the command frame, we use the previously calculated tf
+    if (cmd.header.frame_id.empty() || cmd.header.frame_id == parameters_.robot_link_command_frame)
+    {
+      translation_vector = tf_moveit_to_cmd_frame_.linear() * translation_vector;
```

Comment:
Makes sense to split out these redundant computations. To me it just seems really confusing that `tf_planning_to_cmd_frame` and `tf_moveit_to_cmd_frame` actually have the same parent `planning_frame` but different command frames (either `robot_link_command_frame` or dynamic `cmd_frame` from message). Maybe we could rename the variables somehow to better describe the actual transforms.

Reply:
Agree, changed to `tf_moveit_to_robot_cmd_frame_` and `tf_moveit_to_incoming_cmd_frame` where the second is the one used for incoming commands not in the default command frame